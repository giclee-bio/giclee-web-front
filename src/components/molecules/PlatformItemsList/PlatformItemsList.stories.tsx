import { Meta, StoryObj } from '@storybook/react'

import PlatformItemsList from './PlatformItemsList'

const meta: Meta<typeof PlatformItemsList> = {}

export default meta
type Story = StoryObj<typeof PlatformItemsList>

export const Index: Story = {
  args: {},
}
