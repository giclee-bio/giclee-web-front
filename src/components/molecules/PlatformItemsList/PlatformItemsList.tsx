'use client'

import * as ScrollArea from '@radix-ui/react-scroll-area'

import PlatformItem from '@/components/atoms/PlatformItem'

import type { PlatformItemProps } from '@/components/atoms/PlatformItem'

type Props = {
  items: PlatformItemProps[]
}

const PlatformItemsList: React.FC<Props> = ({ items }) => {
  return (
    <ScrollArea.Root className='h-32 max-w-md overflow-hidden'>
      <ScrollArea.Viewport className='h-full w-full rounded'>
        <div className='flex'>
          {items.map((item, key) => (
            <li className='ml-6 list-none first:ml-8' key={key}>
              <PlatformItem {...item} />
            </li>
          ))}
        </div>
      </ScrollArea.Viewport>
      <ScrollArea.Scrollbar
        className='flex touch-none select-none rounded-b bg-stone-100 p-0.5 transition-colors duration-[160ms] ease-out hover:bg-stone-200 data-[orientation=horizontal]:h-2.5 data-[orientation=vertical]:w-2.5 data-[orientation=horizontal]:flex-col'
        orientation='horizontal'
      >
        <ScrollArea.Thumb className="relative flex-1 rounded-full bg-stone-400 before:absolute before:left-1/2 before:top-1/2 before:h-full before:min-h-[44px] before:w-full before:min-w-[44px] before:-translate-x-1/2 before:-translate-y-1/2 before:content-['']" />
      </ScrollArea.Scrollbar>
    </ScrollArea.Root>
  )
}

export default PlatformItemsList
