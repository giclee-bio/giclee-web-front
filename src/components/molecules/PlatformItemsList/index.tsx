import PlatformItemsList from './PlatformItemsList'

export default PlatformItemsList
