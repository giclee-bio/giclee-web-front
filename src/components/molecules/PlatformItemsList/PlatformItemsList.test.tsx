import { render } from '@testing-library/react'

import PlatformItemsList from './PlatformItemsList'

import type { PlatformItems } from '@/components/atoms/PlatformItem'

const item = {
  alt: 'nacal',
  caption: 'nacal',
  src: '/nacal.png',
}
const items: PlatformItems[] = new Array(8).fill(item)

describe('PlatformItemsList', () => {
  it('snapshot', async () => {
    const { asFragment } = render(<PlatformItemsList items={items} />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
