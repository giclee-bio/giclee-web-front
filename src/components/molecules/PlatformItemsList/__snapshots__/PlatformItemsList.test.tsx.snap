// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformItemsList snapshot 1`] = `
<DocumentFragment>
  <div
    class="h-32 max-w-md overflow-hidden"
    dir="ltr"
    style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
  >
    <style>
      [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
    </style>
    <div
      class="h-full w-full rounded"
      data-radix-scroll-area-viewport=""
      style="overflow-x: scroll; overflow-y: hidden;"
    >
      <div
        style="min-width: 100%; display: table;"
      >
        <div
          class="flex"
        >
          <li
            class="ml-6 list-none first:ml-8"
          >
            <button
              class="w-20"
            >
              <div
                class="flex h-20 w-20 items-center justify-center rounded-xl bg-stone-100"
              >
                <img
                  alt="nacal"
                  class="rounded"
                  data-nimg="1"
                  decoding="async"
                  height="48"
                  loading="lazy"
                  src="/_next/image?url=%2Fnacal.png&w=96&q=75"
                  srcset="/_next/image?url=%2Fnacal.png&w=48&q=75 1x, /_next/image?url=%2Fnacal.png&w=96&q=75 2x"
                  style="color: transparent;"
                  width="48"
                />
              </div>
              <p
                class="mt-1 text-center text-sm"
              >
                nacal
              </p>
            </button>
          </li>
          <li
            class="ml-6 list-none first:ml-8"
          >
            <button
              class="w-20"
            >
              <div
                class="flex h-20 w-20 items-center justify-center rounded-xl bg-stone-100"
              >
                <img
                  alt="nacal"
                  class="rounded"
                  data-nimg="1"
                  decoding="async"
                  height="48"
                  loading="lazy"
                  src="/_next/image?url=%2Fnacal.png&w=96&q=75"
                  srcset="/_next/image?url=%2Fnacal.png&w=48&q=75 1x, /_next/image?url=%2Fnacal.png&w=96&q=75 2x"
                  style="color: transparent;"
                  width="48"
                />
              </div>
              <p
                class="mt-1 text-center text-sm"
              >
                nacal
              </p>
            </button>
          </li>
          <li
            class="ml-6 list-none first:ml-8"
          >
            <button
              class="w-20"
            >
              <div
                class="flex h-20 w-20 items-center justify-center rounded-xl bg-stone-100"
              >
                <img
                  alt="nacal"
                  class="rounded"
                  data-nimg="1"
                  decoding="async"
                  height="48"
                  loading="lazy"
                  src="/_next/image?url=%2Fnacal.png&w=96&q=75"
                  srcset="/_next/image?url=%2Fnacal.png&w=48&q=75 1x, /_next/image?url=%2Fnacal.png&w=96&q=75 2x"
                  style="color: transparent;"
                  width="48"
                />
              </div>
              <p
                class="mt-1 text-center text-sm"
              >
                nacal
              </p>
            </button>
          </li>
          <li
            class="ml-6 list-none first:ml-8"
          >
            <button
              class="w-20"
            >
              <div
                class="flex h-20 w-20 items-center justify-center rounded-xl bg-stone-100"
              >
                <img
                  alt="nacal"
                  class="rounded"
                  data-nimg="1"
                  decoding="async"
                  height="48"
                  loading="lazy"
                  src="/_next/image?url=%2Fnacal.png&w=96&q=75"
                  srcset="/_next/image?url=%2Fnacal.png&w=48&q=75 1x, /_next/image?url=%2Fnacal.png&w=96&q=75 2x"
                  style="color: transparent;"
                  width="48"
                />
              </div>
              <p
                class="mt-1 text-center text-sm"
              >
                nacal
              </p>
            </button>
          </li>
          <li
            class="ml-6 list-none first:ml-8"
          >
            <button
              class="w-20"
            >
              <div
                class="flex h-20 w-20 items-center justify-center rounded-xl bg-stone-100"
              >
                <img
                  alt="nacal"
                  class="rounded"
                  data-nimg="1"
                  decoding="async"
                  height="48"
                  loading="lazy"
                  src="/_next/image?url=%2Fnacal.png&w=96&q=75"
                  srcset="/_next/image?url=%2Fnacal.png&w=48&q=75 1x, /_next/image?url=%2Fnacal.png&w=96&q=75 2x"
                  style="color: transparent;"
                  width="48"
                />
              </div>
              <p
                class="mt-1 text-center text-sm"
              >
                nacal
              </p>
            </button>
          </li>
          <li
            class="ml-6 list-none first:ml-8"
          >
            <button
              class="w-20"
            >
              <div
                class="flex h-20 w-20 items-center justify-center rounded-xl bg-stone-100"
              >
                <img
                  alt="nacal"
                  class="rounded"
                  data-nimg="1"
                  decoding="async"
                  height="48"
                  loading="lazy"
                  src="/_next/image?url=%2Fnacal.png&w=96&q=75"
                  srcset="/_next/image?url=%2Fnacal.png&w=48&q=75 1x, /_next/image?url=%2Fnacal.png&w=96&q=75 2x"
                  style="color: transparent;"
                  width="48"
                />
              </div>
              <p
                class="mt-1 text-center text-sm"
              >
                nacal
              </p>
            </button>
          </li>
          <li
            class="ml-6 list-none first:ml-8"
          >
            <button
              class="w-20"
            >
              <div
                class="flex h-20 w-20 items-center justify-center rounded-xl bg-stone-100"
              >
                <img
                  alt="nacal"
                  class="rounded"
                  data-nimg="1"
                  decoding="async"
                  height="48"
                  loading="lazy"
                  src="/_next/image?url=%2Fnacal.png&w=96&q=75"
                  srcset="/_next/image?url=%2Fnacal.png&w=48&q=75 1x, /_next/image?url=%2Fnacal.png&w=96&q=75 2x"
                  style="color: transparent;"
                  width="48"
                />
              </div>
              <p
                class="mt-1 text-center text-sm"
              >
                nacal
              </p>
            </button>
          </li>
          <li
            class="ml-6 list-none first:ml-8"
          >
            <button
              class="w-20"
            >
              <div
                class="flex h-20 w-20 items-center justify-center rounded-xl bg-stone-100"
              >
                <img
                  alt="nacal"
                  class="rounded"
                  data-nimg="1"
                  decoding="async"
                  height="48"
                  loading="lazy"
                  src="/_next/image?url=%2Fnacal.png&w=96&q=75"
                  srcset="/_next/image?url=%2Fnacal.png&w=48&q=75 1x, /_next/image?url=%2Fnacal.png&w=96&q=75 2x"
                  style="color: transparent;"
                  width="48"
                />
              </div>
              <p
                class="mt-1 text-center text-sm"
              >
                nacal
              </p>
            </button>
          </li>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
